### ISSUE TITLE
Crash when re-logging in after signing out (w/ Google provider)

### ISSUE THREAD

**Issue by seenickcode** [supabase-community/supabase-flutter#48]
# Bug report

## Describe the bug

So I am using `signInWithProvider` + Google successfully, mainly using the [example code provided here](https://github.com/phamhieu/supabase-flutter-demo) to implement a splash screen with `SupabaseAuthState` to receive auth events. The problem I have is that when I log out using `Supabase.instance.client.auth.signOut()`, then try to log in again, an exception is thrown. 

Here's the error: 
```
PlatformException (PlatformException(Error, Error while launching https://<redacted>.supabase.co/auth/v1/authorize?provider=google&redirect_to=com.manninglabs.steadycalendar%3A%2F%2Flogin-callback%2F, null, null))
```

Here's the stack trace.

```
[VERBOSE-2:ui_dart_state.cc(209)] Unhandled Exception: PlatformException(Error, Error while launching https://<redacted>.supabase.co/auth/v1/authorize?provider=google&redirect_to=com.manninglabs.steadycalendar%3A%2F%2Flogin-callback%2F, null, null)
#0      StandardMethodCodec.decodeEnvelope
package:flutter/…/services/message_codecs.dart:607
#1      MethodChannel._invokeMethod
package:flutter/…/services/platform_channel.dart:156
<asynchronous suspension>
#2      launch
package:url_launcher/url_launcher.dart:101
<asynchronous suspension>
#3      GoTrueClientSignInProvider.signInWithProvider
package:supabase_flutter/src/supabase_auth.dart:170
<asynchronous suspension>
#4      LoginOptions._signupTapped
package:lesson02/…/login_options/login_options.dart:45
<asynchronous suspension>
```

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Run the app
```
git clone git@github.com:seenickcode/fluttercrashcourse-lessons.git
git clone -b troubleshooting-relogin-crash 
git checkout --track origin/troubleshooting-relogin-crash
cd module03-steadycalendar/lesson02
flutter run
```

Ensure app runs in a simulator/emulator.

2. Tap "Get Started", then "Continue with Google", a webview will appear, sign in with any Google account, tap "Done" to dismiss the webview after signing in (required to maybe a bug on the iOS simulator? not clear if it's `supabase_flutter`)

3. Now tap "Log Out", tap "Get Started", then "Continue with Google", a webview will appear.
 
4. At this point, the **(app will crash)** (see stacktrace I mentioned above)

4. ## Expected behavior

No crashes when re-attempting to login.

## System information

- OS: iOS simulator (iOS 15.0)
- Version of flutter_supabase: `^0.2.9`

## Context

- [I call login here](https://github.com/seenickcode/fluttercrashcourse-lessons/blob/troubleshooting-relogin-crash/module03-steadycalendar/lesson02/lib/screens/login_options/login_options.dart#L45)
- [I handle supabase auth event here](https://github.com/seenickcode/fluttercrashcourse-lessons/blob/troubleshooting-relogin-crash/module03-steadycalendar/lesson02/lib/screens/splash/splash.dart)

**Comment by bdlukaa** [comment]
sounds like an issue with url_launcher

**Comment by seenickcode** [comment]
Agreed..though a lib supabase_flutter wraps. There a number of url_launcher
options passed in by supabase_flutter, that may be complicating the root
cause.

Based on the error, it looks like it’s having a problem handling an HTTP
response from the oauth flow.

**Comment by seenickcode** [comment]
So I'm still coming across this issue with `supabase_flutter` v0.3.0. 

Please note while [this seems to be a Flutter issue](https://github.com/flutter/flutter/issues/103160), it would be ideal to see if someone on the Supabase team could try to reproduce and investigate the issue via the code cited in the [issue](https://github.com/flutter/flutter/issues/103160) as it indeed is most likely impacting many others in terms of developer experience and adoption of Supabase (`supabase_flutter`) for newcomers to Flutter. 

Thank you.

**Comment by dshukertjr** [comment]
@seenickcode Thanks for letting us know. I will take a look and see if there is anything that we can do on our end.

**Comment by Ashwaq3** [comment]
@seenickcode did you manage to get the session info after signing with google. I had a case when i redirect the user to the app after login i get the session is null.

**Comment by dshukertjr** [comment]
Hi @Ashwaq3,
That sounds like a separate issue, but is it related to this one? If it is a separate issue, would you mind opening a separate issue with details of what is happening with your case?

**Comment by seenickcode** [comment]
After working with the Flutter team on [this issue](https://github.com/flutter/flutter/issues/103160) with `url_launcher`, it appears that the crash was [caused by an older version](https://github.com/flutter/flutter/issues/103160#issuecomment-1138637291) of `url_launcher` that Supabase is using (6.1.0 instead of 6.1.2). CC @dshukertjr

**Comment by dshukertjr** [comment]
@seenickcode Thanks for digging into this. We will ship an update for this then!

**Comment by seenickcode** [comment]
Looks like you got this fix in in the latest 0.3.1+1 of this package @dshukertjr? If so we can close this, right?

**Comment by dshukertjr** [comment]
@seenickcode Yup! Thanks for the reminder!

**Comment by dshukertjr** [comment]
@alvinsga Thanks for reporting this. Reopening to track the issue.

**Comment by Ashwaq3** [comment]
> Hi @Ashwaq3, That sounds like a separate issue, but is it related to this one? If it is a separate issue, would you mind opening a separate issue with details of what is happening with your case?

It's look like it is a separated issue .. I managed to solve it

**Comment by seenickcode** [comment]
Stale
